index.js

```diff
@@ -0,0 +1,57 @@
+const fse = require('fs-extra');
+const path = require('path');
+
+const run = config => {
+
+    if (config.directories && Array.isArray(config.directories)) {
+        const promises = config.directories.map(directoryConfig => copyDirectory(directoryConfig));
+
+        return Promise.all(promises)
+            .then(() => {
+                const message = `${config.directories.length} ${config.directories.length === 1 ? 'directory' : 'directories'} processed`;
+                return {
+                    status: 'complete',
+                    message: message
+                };
+            })
+            .catch(error => {
+                return {
+                    status: 'error',
+                    error: error
+                };
+            });
+    }
+
+    return {
+        status: 'error',
+        error: 'Config directories is not found or not an array.'
+    };
+};
+
+const copyDirectory = fileConfig => {
+
+    const directory = fse.existsSync(fileConfig.dest);
```

Comment:
Is it recommended to use synchronous functions to check the existence of a directory and create it?

Reply:
I'd use asynchronous functions whenever you can, so as to not block execution (in case we run tasks in parallel)